lib/procore/auth/client_credentials.rb

```diff
@@ -43,9 +44,13 @@ def client
         OAuth2::Client.new(
           client_id,
           client_secret,
-          site: "#{host}/oauth/token",
+          site: "#{auth_host}/oauth/token",
         )
       end
+
+      def auth_host
+        ::ENV.fetch("PROCORE_AUTH_HOST", nil) || AUTH_HOST
```

Comment:
Wanted to start a conversation here on whether we should require that this is read from an envar, or keep it optional. It seems more user friendly this way to not require gem consumers to have to set this, but I'm open to hearing other options!

Reply:
I think a sane default (like "auth.procore.com") with option to override is great.